frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx

```diff
@@ -23,6 +25,15 @@ export const NotebookContainer = ({
     isOpen && setShouldShowNotebook(isOpen);
   }, [isOpen]);
 
+  const { isNativePreviewSidebarOpen } = useSelector(getUiControls);
+  const [shouldShowCodePreview, setShouldShowCodePreview] = useState(
```

Comment:
There should be no state copy here I think. You can dispatch an action to update `uiControls` directly instead

Reply:
You're absolutely right!
The magic of `useSelector` 🦄 

Btw, the neat part is that we're not even updating the `uiControls` directly from the sidebar. That state is controlled with the button in the QB header.